File: app/src/main/java/so/codeweaver/muzei/ponies/DerpibooruResult.java
package so.codeweaver.muzei.ponies;

/**
 * Created by berwyn on 09/04/15.
 */
public final class DerpibooruResult {

    public int     total;
    public Image[] search;

    public static final class Image {
        public String id;
        public String image;
        public String uploader;
        public String tags;
    }

}

File: app/src/main/java/so/codeweaver/muzei/ponies/derpi/DerpibooruResult.kt
package so.codeweaver.muzei.ponies.derpi

import java.util.Arrays

data class DerpibooruResult(
        val total: Int,
        val search: Array<DerpibooruImage>
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as DerpibooruResult

        if (total != other.total) return false
        if (!Arrays.equals(search, other.search)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = total
        result = 31 * result + Arrays.hashCode(search)
        return result
    }
}
